### PR TITLE
Database : IOfflineProvider - .manifest URL creation

### PR DESCRIPTION
**Pull request by gabrielheming**
Issue #10278

### CODE REVIEW

File: src/Offline/database.ts [hunk 1]
```diff
@@ -108,16 +108,17 @@ export class Database implements IOfflineProvider {
         };
 
         var timeStampUsed = false;
-        var manifestURL = this._currentSceneUrl + ".manifest";
+        var manifestURL = new URL(this._currentSceneUrl)
```
Comment (RaananW) [inline src/Offline/database.ts:111]: you will still need to check if `typeof URL !== 'undefined'`. Something like this:

```suggestion
        const manifestURL = typeof URL !== 'undefined' ? new URL(this._currentSceneUrl) : this._currentSceneUrl + ".manifest";
```

Of course the rest of the changes will need to react correctly to that as well. So maybe an `if( typeof URL !== 'undefined')` will work better

**Review by RaananW — CHANGES_REQUESTED** [review]
Sadly we still do need to support browsers with no URL element, but it is ok if the previous functionality will still be used on older browsers.

File: src/Offline/database.ts (hunk 1 above)
Comment (gabrielheming) [reply src/Offline/database.ts]: Agreed. I've refactored into an internal function to avoid duplication.

**Comment by RaananW** [comment]
Linting fails. Is timestamp still supported with this method? (looks like it, i am just making sure)

**Comment by gabrielheming** [comment]
It should work properly. It returns a regular string.

The part of timestamp, I've reverted to the original, I am not seeing any lint error. Thoug, I've seen on `Date.now()` that should be `Date.now().toString()`

**Comment by RaananW** [comment]
`Date.now()` returns a number, string + number will result in a string, so it's all good.

**Comment by gabrielheming** [comment]
> `Failed to lint: /home/vsts/work/1/s/src/Offline/database.ts [118, 10]: Missing semicolon, /home/vsts/work/1/s/src/Offline/database.ts [112, 57]: Missing semicolon.`

Ops, my bad. It fixed.

> `Date.now()` returns a number, string + number will result in a string, so it's all good.

Yep, indeed. It was just showing it here before.

**Comment by RaananW** [comment]
thank you so much for this! :-)

**Pull request merged**